tests/integration/kubernetes/k8s-confidential.bats

```diff
@@ -10,17 +10,22 @@ load "${BATS_TEST_DIRNAME}/confidential_common.sh"
 load "${BATS_TEST_DIRNAME}/tests_common.sh"
 
 setup() {
-	SUPPORTED_HYPERVISORS=("qemu-sev" "qemu-snp" "qemu-tdx" "qemu-se")
+	SUPPORTED_TEE_HYPERVISORS=("qemu-sev" "qemu-snp" "qemu-tdx" "qemu-se")
+	SUPPORTED_NON_TEE_HYPERVISORS=("qemu")
 
 	# This check must be done with "<SPACE>${KATA_HYPERVISOR}<SPACE>" to avoid
 	# having substrings, like qemu, being matched with qemu-$something.
-	[[ " ${SUPPORTED_HYPERVISORS[*]} " =~  " ${KATA_HYPERVISOR} " ]] ||  skip "Test not supported for ${KATA_HYPERVISOR}."
+	if [[ " ${SUPPORTED_TEE_HYPERVISORS[*]} " =~  " ${KATA_HYPERVISOR} " ]] || [[ " ${SUPPORTED_NON_TEE_HYPERVISORS[*]} " =~  " ${KATA_HYPERVISOR} " ]]; then
+		skip "Test not supported for ${KATA_HYPERVISOR}."
+	fi
 
 	get_pod_config_dir
```

Comment:
I'd add an if-else here to already call the method to set-up annotations, as that just got merged this morning

Reply:
@fidencio thanks for the feedback so I just have a question, for the supported_tee_hypervisors I need to call https://github.com/kata-containers/kata-containers/blob/main/tests/integration/kubernetes/setup.sh#L100 but in the case of qemu that does not apply

Reply:
Actually the opposite, @GabyCT.

For the **TEEs** we don't need to set the annotation, because those will be using the correct image (the confidential image)
For the **non-TEEs** we need to set the annotation, because for those we want to use the confidential image instead of the normal one).

We can even split that function out of the setup.sh and put it somewhere easier to be accessed.

Does the explanation help?